### PR TITLE
OnExtendedCentreline for 2.5 NM

### PR DESCRIPTION
**Pull request by catcarbon**
### Demo
https://github.com/mmp/vice/assets/9087266/d5e185f6-1d46-4846-b79a-c8f8f3241543

### CODE REVIEW

File: nav.go [hunk 1]
```diff
@@ -307,13 +307,17 @@ func (nav *Nav) OnApproach(checkAltitude bool) bool {
 	return false
 }
 
-func (nav *Nav) OnFinalApproach() bool {
-	if !nav.OnApproach(false) {
-		return false
-	}
+// OnExtendedCenterline checks if the flight position is less than maxNmDeviation
+// from the infinite line defined by the assigned approach localizer
+func (nav *Nav) OnExtendedCenterline(maxNmDeviation float32) bool {
```
Comment (catcarbon) [inline nav.go:312]: Create `OnExtendedCenterline` to determine if the aircraft looks like it's on the final approach course. Stole the idea from `TurningToJoin` section in `LocalierHeading`.

File: airport.go [hunk 2]
```diff
@@ -438,8 +438,8 @@ func (ap *Airport) PostDeserialize(icao string, sg *ScenarioGroup, e *ErrorLogge
 			e.ErrorString("exit not in scenario group \"scratchpads\"")
 		}
 
-		if dep.Altitude < 500 && dep.Altitude != 0{
-			e.ErrorString("altitude of %v is too low to be used. Is it supposed to be %v?",dep.Altitude, dep.Altitude*100)
+		if dep.Altitude < 500 && dep.Altitude != 0 {
+			e.ErrorString("altitude of %v is too low to be used. Is it supposed to be %v?", dep.Altitude, dep.Altitude*100)
```
Comment (catcarbon) [inline airport.go:442]: Looks like my editor auto-applied styling on merge.

File: nav.go [hunk 3]
```diff
@@ -905,10 +909,7 @@ func (nav *Nav) LocalizerHeading(wind WindModel, lg *Logger) (heading float32, t
 
 	case TurningToJoin:
 		// we've turned to intercept. have we intercepted?
-		loc := ap.Line()
-		dist := PointLineDistance(ll2nm(nav.FlightState.Position, nav.FlightState.NmPerLongitude),
-			ll2nm(loc[0], nav.FlightState.NmPerLongitude), ll2nm(loc[1], nav.FlightState.NmPerLongitude))
-		if dist > .2 {
+		if !nav.OnExtendedCenterline(.2) {
```
Comment (catcarbon) [inline nav.go:912]: Replace with the new method.

File: nav.go [hunk 4]
```diff
@@ -1367,42 +1368,33 @@ func (nav *Nav) getUpcomingSpeedRestrictionWaypoint() (*Waypoint, float32, float
 }
 
 // distanceToEndOfApproach returns the remaining distance to the last
-// waypoint, and a flag indicating if the FAF has passed for an
-// aircraft that has been given an approach.
-func (nav *Nav) distanceToEndOfApproach() (float32, bool, error) {
+// waypoint (usually runway threshold) of the currently assigned approach.
+func (nav *Nav) distanceToEndOfApproach() (float32, error) {
```
Comment (catcarbon) [inline nav.go:1372]: Remove logic that checks if the aircraft has passed FAF (no longer used and doesn't look useful).

File: stars.go [hunk 5]
```diff
@@ -5846,7 +5846,10 @@ func (sp *STARSPane) checkInTrailCwtSeparation(back, front *Aircraft) {
 			nmdistance2ll(vol.Threshold, state.TrackPosition()) < vol.Dist25nmApproach {
 
 			// between aircraft established on the final approach course
-			if back.OnFinalApproach() && front.OnFinalApproach() {
+			// Note 1: checked with OnExtendedCenterline since reduced separation probably
+			// doesn't apply to approaches with curved final approach segment
+			// Note 2: 0.2 NM is slightly less than full-scale deflection at 5 NM out
+			if back.OnExtendedCenterline(.2) && front.OnExtendedCenterline(.2) {
```
Comment (catcarbon) [inline stars.go:5852]: Change method and update note.

File: airport.go (hunk 2 above)
Comment (mmp) [reply airport.go:442]: Yep, all good. I need to add a pre-merge check to make sure nothing changes after a gofmt run to make sure things are formatted when they land. (I have my editor set up to do this as well.)

File: nav.go (hunk 1 above)
Comment (mmp) [reply nav.go:312]: This is great to have as a stand-alone method.

File: stars.go (hunk 5 above)
Comment (mmp) [reply stars.go:5852]: (The 0.2nm was very ad-hoc so I'm more than open to a more principled test for this, maybe in the future removing the parameter to `OnExetendedCenterline` and baking in a distance-based criterion (aka angle-based w.r.t. the threshold point) instead.

**Review by mmp — APPROVED** [review]

**Pull request merged**